### PR TITLE
Add See also section to vn entropy and fidelity

### PR DESCRIPTION
**Pull request by rmoyard**
**Description of the Change:**

It adds some see also sections for entropy and fidelity.

**Related GitHub Issues:**

Closes https://github.com/PennyLaneAI/pennylane/issues/2764

### CODE REVIEW

**Comment by eddddddy** [comment]
Great change @rmoyard 🔥 🔥 🔥 ! 

While we're adding cross-references, can you also add them for the various `density_matrix` and `mutual_info` functions?

**Comment by rmoyard** [comment]
Yes for sure 👍

**Comment by eddddddy** [comment]
One of the mutual info links isn't rendering correctly: https://pennylane--2775.org.readthedocs.build/en/2775/code/api/pennylane.qinfo.transforms.mutual_info.html

File: pennylane/qinfo/transforms.py [hunk 1]
```diff
@@ -171,9 +173,7 @@ def circuit(x):
     >>> qml.grad(mutual_info_circuit)(np.array(0.4, requires_grad=True))
     1.2430067731198946
 
-    .. seealso::
-
-        :func:`~.qinfo.vn_entropy`
+    .. seealso:: :func:`~.qinfo.vn_entropy`, `pennylane.math.mutual_info` and :func:`pennylane.mutual_info`
```
Comment (rmoyard) [inline pennylane/qinfo/transforms.py:176]:
```suggestion
    .. seealso:: :func:`~.qinfo.vn_entropy`, :func:`pennylane.math.mutual_info` and :func:`pennylane.mutual_info`
```

**Pull request merged**